src/mpc-hc/SubtitlesProvidersUtils.cpp

```diff
@@ -0,0 +1,781 @@
+{
+    std::string result;
+
+    UINT buffer_len = 32 * 1024;
+    std::vector<BYTE> buffer(buffer_len);
+
+    z_stream deflate_stream = { 0 };
+    deflate_stream.next_in = (BYTE*)data.c_str();
+    deflate_stream.avail_in = (UINT)data.length();
+
+    int ret = Z_OK;
+    //if ((ret = deflateInit2(&deflate_stream, Z_BEST_COMPRESSION, Z_DEFLATED, DEF_WBITS, DEF_MEM_LEVEL, Z_DEFAULT_STRATEGY)) == Z_OK) {
+    if ((ret = deflateInit(&deflate_stream, Z_BEST_COMPRESSION)) == Z_OK) {
+        do {
+            deflate_stream.next_out = &buffer[0];
+            deflate_stream.avail_out = buffer_len;
+            if ((ret = deflate(&deflate_stream, Z_FINISH)) >= Z_OK) {
+                result.append((char*)&buffer[0], buffer_len - deflate_stream.avail_out);
+            } else { break; }
+        } while (deflate_stream.avail_out == 0);
+        ret = deflateEnd(&deflate_stream);
+    }
+    return result;
+}
+
+std::string string_gzinflate(const std::string& data)
+{
+    std::string result;
+
+    UINT buffer_len = 32 * 1024;
+    std::vector<BYTE> buffer(buffer_len);
+
+    z_stream inflate_stream = { 0 };
+    inflate_stream.next_in = (BYTE*)data.c_str();
+    inflate_stream.avail_in = (UINT)data.length();
+
+    int ret = Z_OK;
+    if ((ret = inflateInit2(&inflate_stream, DEF_WBITS + Z_DECODING_ZLIB_GZIP)) == Z_OK) {
+        do {
+            inflate_stream.next_out = &buffer[0];
+            inflate_stream.avail_out = buffer_len;
+            if ((ret = inflate(&inflate_stream, Z_NO_FLUSH)) >= Z_OK) {
+                result.append((char*)&buffer[0], buffer_len - inflate_stream.avail_out);
+            } else { break; }
+        } while (inflate_stream.avail_out == 0);
+        ret = inflateEnd(&inflate_stream);
+    }
+    return result;
+}
+
+std::string string_gzuncompress(const std::string& data)
+{
+    std::string result;
+
+    UINT buffer_len = 32 * 1024;
+    std::vector<BYTE> buffer(buffer_len);
+
+    z_stream inflate_stream = { 0 };
+    inflate_stream.next_in = (BYTE*)data.c_str();
+    inflate_stream.avail_in = (UINT)data.length();
+
+    int ret = Z_OK;
+    //if ((ret = inflateInit2(&inflate_stream, DEF_WBITS)) == Z_OK) {
+    if ((ret = inflateInit(&inflate_stream)) == Z_OK) {
+        do {
+            inflate_stream.next_out = &buffer[0];
+            inflate_stream.avail_out = buffer_len;
+            if ((ret = inflate(&inflate_stream, Z_NO_FLUSH)) >= Z_OK) {
+                result.append((char*)&buffer[0], buffer_len - inflate_stream.avail_out);
+            } else { break; }
+        } while (inflate_stream.avail_out == 0);
+        ret = inflateEnd(&inflate_stream);
+    }
+    return result;
+}
+
+int file_unzip(CStringA file, CStringA fn, string_map& dataOut)
+{
+#define dir_delimter '/'
+#define MAX_FILENAME 512
+#define READ_SIZE 8192
+    // Open the zip file
+    unzFile zipfile = unzOpen(file);
+    if (zipfile == nullptr) {
+        printf("%s: not found\n");
```

Comment:
This (and others) should be changed to `TRACE` or something. Also string is missing in this case.